File: examples/js/basic-image.ts
/**
 * Basic Image Generation Example
 *
 * This example demonstrates how to generate images using the Oaktis SDK.
 */

import { OaktisClient } from '@oaktis/sdk';

async function main() {
  // Initialize the client with your API key
  const client = new OaktisClient({
    apiKey: process.env.OAKTIS_API_KEY!,
  });

  try {
    console.log('🖼️  Generating images...\n');

    // Generate images
    const job = await client.image.generate({
      prompt: 'a futuristic city at night with neon lights',
      size: '1024x1024',
      n: 2,
    });

    console.log('✅ Job submitted successfully!');
    console.log(`Job ID: ${job.id}`);
    console.log(`Status: ${job.status}\n`);

    // Poll for completion
    console.log('⏳ Waiting for completion...\n');

    let status = await client.image.getStatus(job.id);
    while (status.status === 'pending' || status.status === 'processing') {
      if (status.progress !== undefined) {
        console.log(`Progress: ${status.progress}%`);
      }

      await new Promise((resolve) => setTimeout(resolve, 2000));
      status = await client.image.getStatus(job.id);
    }

    if (status.status === 'completed') {
      const result = await client.image.getJob(job.id);
      console.log('\n✅ Images generated successfully!');
      if (result.imageUrls && result.imageUrls.length > 0) {
        console.log(`\nGenerated ${result.imageUrls.length} image(s):`);
        result.imageUrls.forEach((url, i) => {
          console.log(`  ${i + 1}. ${url}`);
        });
      }
    } else {
      console.error('\n❌ Image generation failed:', status.error);
      process.exit(1);
    }
  } catch (error: any) {
    console.error('Error:', error.message || error);
    process.exit(1);
  }
}

main();

File: examples/python/basic_image.py
"""
Basic Image Generation Example

This example demonstrates how to generate images using the Oaktis SDK.
"""

import asyncio
import os
from oaktis import OaktisClient, ImageGenerateParams


async def main():
    # Initialize the client with your API key
    client = OaktisClient(api_key=os.getenv("OAKTIS_API_KEY"))

    try:
        print("🖼️  Generating images...\n")

        # Generate images
        job = await client.image.generate(
            ImageGenerateParams(
                prompt="a futuristic city at night with neon lights",
                size="1024x1024",
                n=2,
            )
        )

        print("✅ Job submitted successfully!")
        print(f"Job ID: {job.id}")
        print(f"Status: {job.status}\n")

        # Poll for completion
        print("⏳ Waiting for completion...\n")

        while True:
            status = await client.image.get_status(job.id)

            if status.progress is not None:
                print(f"Progress: {status.progress}%")

            if status.status in ["completed", "failed"]:
                break

            await asyncio.sleep(2)

        if status.status == "completed":
            result = await client.image.get_job(job.id)
            print("\n✅ Images generated successfully!")
            if result.image_urls and len(result.image_urls) > 0:
                print(f"\nGenerated {len(result.image_urls)} image(s):")
                for i, url in enumerate(result.image_urls, 1):
                    print(f"  {i}. {url}")
        else:
            print(f"\n❌ Image generation failed: {status.error}")
            exit(1)

    except Exception as e:
        print(f"Error: {str(e)}")
        exit(1)
    finally:
        await client.close()


if __name__ == "__main__":
    asyncio.run(main())
